shell/platform/windows/windows_lifecycle_manager.cc

```diff
@@ -0,0 +1,151 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "windows_lifecycle_manager.h"
+
+#include <TlHelp32.h>
+#include <WinUser.h>
+#include <Windows.h>
+#include <tchar.h>
+
+#include "flutter/shell/platform/windows/flutter_windows_engine.h"
+
+namespace flutter {
+
+WindowsLifecycleManager::WindowsLifecycleManager(FlutterWindowsEngine* engine)
+    : engine_(engine) {}
+
+WindowsLifecycleManager::~WindowsLifecycleManager() {}
+
+void WindowsLifecycleManager::Quit(std::optional<HWND> hwnd,
+                                   std::optional<WPARAM> wparam,
+                                   std::optional<LPARAM> lparam,
+                                   UINT exit_code) {
+  if (!hwnd.has_value()) {
+    ::PostQuitMessage(exit_code);
+  } else {
+    BASE_CHECK(wparam.has_value() && lparam.has_value());
+    sent_close_messages_[*hwnd]++;
```

Comment:
We should only ignore `WM_CLOSE` messages whose `wparam` and `lparam` matches what we posted.